pkg/controller/certificatesigningrequest/ipsec_csr_approver.go

```diff
@@ -0,0 +1,158 @@
+		"system", "serviceaccount", env.GetAntreaNamespace(), "antrea-agent",
+	}, ":")
+)
+
+type ipsecCSRApprover struct {
+	client clientset.Interface
+}
+
+var ipsecTunnelUsages = sets.NewString(
+	string(certificatesv1.UsageIPsecTunnel),
+)
+
+var _ approver = (*ipsecCSRApprover)(nil)
+
+func (ic *ipsecCSRApprover) recognize(csr *certificatesv1.CertificateSigningRequest) bool {
+	if csr.Spec.SignerName != antreaapis.AntreaIPsecCSRSignerName {
+		return false
+	}
+	for _, usage := range csr.Spec.Usages {
+		if !ipsecTunnelUsages.Has(string(usage)) {
+			return false
+		}
+	}
+	return true
+}
+
+func (ic *ipsecCSRApprover) verify(csr *certificatesv1.CertificateSigningRequest) (bool, error) {
+	var failedReasons []string
+	cr, err := decodeCertificateRequest(csr.Spec.Request)
+	if err != nil {
+		return false, err
+	}
+	if err := ic.verifyCertificateRequest(cr, csr.Spec.Usages); err != nil {
+		if _, ok := err.(*transientError); ok {
+			return false, err
+		} else {
+			failedReasons = append(failedReasons, err.Error())
+		}
+	}
+	if err := ic.verifyIdentity(cr.Subject.CommonName, csr); err != nil {
+		if _, ok := err.(*transientError); ok {
+			return false, err
+		}
+		failedReasons = append(failedReasons, err.Error())
+	}
+
+	if len(failedReasons) > 0 {
+		klog.InfoS("Verify CertificateSigningRequest for IPsec failed", "reasons", failedReasons, "CSR", csr.Name)
+		return false, nil
+	}
+	return true, nil
+}
+
+func (ic *ipsecCSRApprover) name() string {
+	return ipsecCSRApproverName
+}
+
+func (ic *ipsecCSRApprover) verifyCertificateRequest(req *x509.CertificateRequest, usages []certificatesv1.KeyUsage) error {
+	if !reflect.DeepEqual(req.Subject.Organization, []string{antreaapis.AntreaOrganizationName}) {
+		return errOrganizationNotAntrea
+	}
+	if req.Subject.CommonName == "" {
+		return errCommonNameRequired
+	}
+	if len(req.URIs) > 0 {
+		return errURISANNotAllowed
+	}
+	if len(req.IPAddresses) > 0 {
+		return errIPSANNotAllowed
+	}
+	if len(req.EmailAddresses) > 0 {
+		return errEmailSANNotAllowed
+	}
+	if !reflect.DeepEqual([]string{req.Subject.CommonName}, req.DNSNames) {
+		return errDNSSANNotMatchCommonName
+	}
+	for _, u := range usages {
+		if !ipsecTunnelUsages.Has(string(u)) {
+			return fmt.Errorf("unsupported key usage: %v", u)
+		}
+	}
+	_, err := ic.client.CoreV1().Nodes().Get(context.TODO(), req.Subject.CommonName, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		return fmt.Errorf("requested Node %s not found", req.Subject.CommonName)
+	} else if err != nil {
+		return &transientError{err}
+	}
+	return nil
+}
+
+func (ic *ipsecCSRApprover) verifyIdentity(nodeName string, csr *certificatesv1.CertificateSigningRequest) error {
+	if csr.Spec.Username != antreaAgentServiceAccountName {
+		return errUserUnrecognized
+	}
+	podNameValues, podUIDValues := csr.Spec.Extra[sautil.PodNameKey], csr.Spec.Extra[sautil.PodUIDKey]
+	if len(podNameValues) == 0 || len(podUIDValues) == 0 {
+		return errExtraFieldsRequired
```

Comment:
don't we get a conclusion that podName validation is nice to have and it could still approve it but log a warning?
I'm fine with not approving it automatically but it should still log a warning about how to fix it.

Reply:
added a warning log here.